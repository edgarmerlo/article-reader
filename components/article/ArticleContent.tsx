import React from 'react';
import Paper from '@mui/material/Paper';

interface ArticleContentProps {
  children: React.ReactNode;
}

export const ArticleContent: React.FunctionComponent<ArticleContentProps> = ({
  children,
}: ArticleContentProps) => {

  return (
    <Paper
      elevation={0}
      sx={{
        marginTop: 2,
        marginBottom: 2
      }}
    >
      {children}
    </Paper>
  );
};
