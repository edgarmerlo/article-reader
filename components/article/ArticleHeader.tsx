import React from 'react';
import CardHeader from '@mui/material/CardHeader';
import Avatar from '@mui/material/Avatar';
import { Typography } from '@mui/material';
import { DateTimeService } from 'services/DateTimeService';

interface ArticleHeaderProps {
  title: string,
  author: string,
  date: string
}

export const ArticleHeader: React.FunctionComponent<ArticleHeaderProps> = ({
  title,
  author,
  date
}: ArticleHeaderProps) => {

  return (
    <>
      <Typography variant="h3" component="h1">
        {title}
      </Typography>
      <CardHeader
        avatar={
          <Avatar aria-label="recipe">
            {author.charAt(0)}
          </Avatar>
        }
        title={`Author: ${author}`}
        subheader={DateTimeService.getDateFromString(date)}
      />
    </>
  );
};
