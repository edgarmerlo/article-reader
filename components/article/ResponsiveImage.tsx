import React from 'react';
import styles from 'styles/components/responsiveImage.module.scss';
import { baseUrl } from 'configs';

interface ResponsiveImageProps {
  imageUrl: string;
}

export const ResponsiveImage: React.FunctionComponent<ResponsiveImageProps> = ({
  imageUrl,
}: ResponsiveImageProps) => {

  return (
    <img src={`${baseUrl}${imageUrl}`} className={styles.image} />
  );
};
