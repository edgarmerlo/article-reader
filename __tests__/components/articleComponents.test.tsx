import { render } from '@testing-library/react';
import { ArticleContent } from 'components/article/ArticleContent';
import { ArticleHeader } from 'components/article/ArticleHeader';
import { ResponsiveImage } from 'components/article/ResponsiveImage';

describe('Snapshots for article components', () => {
  it('renders articlecontent', () => {
    const { container } = render(<ArticleContent >Hi</ArticleContent>);
    expect(container).toMatchSnapshot();
  });

  it('renders article header', () => {
    const { container } = render(<ArticleHeader author='Test' title='test' date='2020-12-09' />);
    expect(container).toMatchSnapshot();
  });

  it('renders article header', () => {
    const { container } = render(<ResponsiveImage imageUrl='/test' />);
    expect(container).toMatchSnapshot();
  });
});