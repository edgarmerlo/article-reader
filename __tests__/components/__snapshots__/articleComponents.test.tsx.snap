// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles service renders homepage unchanged 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-61ve4i-MuiPaper-root"
  >
    Hi
  </div>
</div>
`;

exports[`Snapshots for article components renders article header 1`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
  >
    test
  </h1>
  <div
    class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
    >
      <div
        aria-label="recipe"
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
      >
        T
      </div>
    </div>
    <div
      class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <span
        class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
      >
        Author: Test
      </span>
      <span
        class="MuiTypography-root MuiTypography-body2 MuiCardHeader-subheader css-83ijpv-MuiTypography-root"
      >
        08 / 12 / 2020
      </span>
    </div>
  </div>
</div>
`;

exports[`Snapshots for article components renders article header 2`] = `
<div>
  <img
    class="image"
    src="https://content-eu-4.content-cms.com//test"
  />
</div>
`;

exports[`Snapshots for article components renders articlecontent 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-61ve4i-MuiPaper-root"
  >
    Hi
  </div>
</div>
`;
